File: src/web/lineup.js
const channels = require('../../channels.json');

const lineup = (req, res) => {
  res.setHeader('Content-Type', 'application/json');
  res.send(
    JSON.stringify(
      channels.map((channel, index) => ({
        GuideNumber: (index + 1).toString(),
        GuideName: channel.name,
        Tags: [],
        URL: `http://${req.headers.host}/stream/${index + 1}`,
      })),
    ),
  );
};

const lineupStatus = (req, res) => {
  res.setHeader('Content-Type', 'application/json');
  res.send(
    JSON.stringify(
      {
        ScanInProgress: 0,
        ScanPossible: 1,
        Source: 'Cable',
        SourceList: ['Cable'],
      },
    ),
  );
};

module.exports = {
  lineup,
  lineupStatus,
};

File: routes/lineup.go
package routes

import (
	"fmt"
	"net/http"

	"plex-dvr-hls/config"
	"github.com/gin-gonic/gin"
)

type ChannelLineup struct {
	GuideNumber string   `json:"GuideNumber"`
	GuideName   string   `json:"GuideName"`
	Tags        []string `json:"Tags"`
	URL         string   `json:"URL"`
}

func Lineup(c *gin.Context) {
	var channelLineups []ChannelLineup

	var host = c.Request.Host

	for index, channel := range config.Channels {
		channelLineups = append(
			channelLineups,
			ChannelLineup{
				GuideNumber: fmt.Sprintf("%d", index+1),
				GuideName:   channel.Name,
				Tags:        make([]string, 0),
				URL:         fmt.Sprintf("http://%s/stream/%d", host, index+1),
			},
		)
	}

	c.JSON(
		http.StatusOK,
		channelLineups,
	)
}

type Status struct {
	ScanInProgress int      `json:"ScanInProgress"`
	ScanPossible   int      `json:"ScanPossible"`
	Source         string   `json:"Source"`
	SourceList     []string `json:"Cable"`
}

func LineupStatus(c *gin.Context) {
	c.JSON(
		http.StatusOK,
		Status{
			ScanInProgress: 0,
			ScanPossible:   1,
			Source:         "Cable",
			SourceList: []string{
				"Cable",
			},
		},
	)

}
